task2_Yiska_Levi/src/Board.cpp

```diff
@@ -0,0 +1,293 @@
+#include "Board.h"
+#include "GeneralTool.h" // Include the header file for GeneralTool
+#include "Rook.h" // Include the header file for GeneralTool
+#include "King.h" // Include the header file for GeneralTool
+#include "Pawn.h"
+#include "Bishop.h"
+#include "Knight.h"
+#include "Queen.h"
+#include <iostream>
+
+Board::Board() {
+    // Initialize the board with nullptrs
+    for (int i = 0; i < 8; i++) {
+        for (int j = 0; j < 8; j++) {
+            _board[i][j] = nullptr;
+        }
+    }
+    initialize();
+}
+
+Board::Board(const Board& other) {
+    // Copy the state of the other board
+    _white_turn = other._white_turn;
+
+    // Copy the tools on the board
+    for (int i = 0; i < 8; ++i) {
+        for (int j = 0; j < 8; ++j) {
+            GeneralTool* tool = other._board[i][j];
+            if (tool != nullptr) {
+                _board[i][j] = tool; // Assuming GeneralTool has a virtual clone() function
+            }
+            else {
+                _board[i][j] = nullptr;
+            }
+        }
+    }
+}
+
+Board::~Board() {
+    // Clean up the board by deleting the tools
+    for (int i = 0; i < 8; i++) {
+        for (int j = 0; j < 8; j++) {
+            if(_board[i][j] != nullptr)
+                delete _board[i][j];
+        }
+    }
+}
+
+void Board::initialize() {
+    // Create and place the tools in their starting positions
+
+// Place the pawns
```

Comment:
creating the pieces manually and not by the string passed to the chess class
if i change the string for testing i can play with empty pieces